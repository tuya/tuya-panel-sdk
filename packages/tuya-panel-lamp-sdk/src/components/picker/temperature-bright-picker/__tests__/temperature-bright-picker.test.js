import React from 'react';
import { shallow, mount } from 'enzyme';
import TemperatureBrightPicker from '../index';

describe('TemperatureBrightPicker components', () => {
  const origConsole = console.error;
  beforeEach(() => {
    console.error = () => {};
  });
  afterEach(() => {
    console.error = origConsole;
  });

  it('basic render', () => {
    let value = { brightness: 1000, temperature: 1000 };
    const wrapper = shallow(
      <TemperatureBrightPicker value={{ brightness: 1000, temperature: 1000 }} />
    );
    expect(wrapper).toMatchSnapshot();

    value = { brightness: 500, temperature: 500 };
    wrapper.setProps({ value });
  });

  it('componentWillReceiveProps render', () => {
    let disabled = false;
    let hideThumb = false;

    const wrapper = shallow(<TemperatureBrightPicker disabled={disabled} hideThumb={hideThumb} />);
    expect(wrapper).toMatchSnapshot();

    disabled = true;
    wrapper.setProps({ disabled });

    hideThumb = true;
    wrapper.setProps({ hideThumb });

    wrapper.setProps({
      renderThumb: () => {
        return <View />;
      },
    });

    wrapper.unmount();
  });

  it('method', () => {
    const wrapper = shallow(<TemperatureBrightPicker />);
    expect(wrapper).toMatchSnapshot();
    wrapper.instance().handleGrant(500);
    wrapper.instance().handleMove(500);
    wrapper.instance().handleRelease(500);
    wrapper.instance().handlePress(500);
    wrapper.instance().handleChange(500);
    wrapper.instance().handleChangeBright(500);
    wrapper.unmount();
  });
});
