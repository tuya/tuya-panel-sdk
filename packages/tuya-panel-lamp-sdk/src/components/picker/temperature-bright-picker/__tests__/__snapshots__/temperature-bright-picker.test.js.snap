// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemperatureBrightPicker components basic render 1`] = `
<View
  maxBrightPercent={100}
  maxBrightness={1000}
  minBrightPercent={1}
  minBrightness={10}
  onChange={[Function]}
  onGrant={[Function]}
  onMove={[Function]}
  onPress={[Function]}
  onRelease={[Function]}
  stepBrightPercent={1}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <TemperaturePicker
    accessibilityLabel="TemperaturePicker"
    disabled={false}
    disalbedThumbOpacity={0.4}
    gradientBg={
      Array [
        Object {
          "offset": "0%",
          "stopColor": "#FFCA5C",
          "stopOpacity": 1,
        },
        Object {
          "offset": "50%",
          "stopColor": "#fff",
          "stopOpacity": 1,
        },
        Object {
          "offset": "100%",
          "stopColor": "#CDECFE",
          "stopOpacity": 1,
        },
      ]
    }
    hideThumb={false}
    innerRadius={132}
    max={1000}
    min={0}
    offsetAngle={180}
    onChange={[Function]}
    onGrant={[Function]}
    onMove={[Function]}
    onPress={[Function]}
    onRelease={[Function]}
    opacityAnimationDuration={150}
    opacityAnimationValue={1}
    radius={270}
    sideOffset={10}
    step={1}
    storageKey="temperature_bright_picker"
    thumbRadius={32}
    touchOffset={0}
    touchThumbRadius={0}
    value={1000}
  />
  <View
    style={
      Object {
        "height": 250.27984337537052,
        "justifyContent": "center",
        "position": "absolute",
        "width": 84,
      }
    }
  >
    <ThemedPickerView
      itemStyle={
        Object {
          "color": "#fff",
          "fontSize": 42,
          "height": 250.27984337537052,
        }
      }
      itemTextColor="#fff"
      onValueChange={[Function]}
      selectedItemTextColor="#fff"
      selectedValue="100"
      style={
        Object {
          "height": 250.27984337537052,
        }
      }
      textSize={42}
      theme={Object {}}
      useTYRCTPicker={false}
      visibleItemCount={3}
    >
      <PickerIOSItem
        key="1"
        label="1"
        value="1"
      />
      <PickerIOSItem
        key="2"
        label="2"
        value="2"
      />
      <PickerIOSItem
        key="3"
        label="3"
        value="3"
      />
      <PickerIOSItem
        key="4"
        label="4"
        value="4"
      />
      <PickerIOSItem
        key="5"
        label="5"
        value="5"
      />
      <PickerIOSItem
        key="6"
        label="6"
        value="6"
      />
      <PickerIOSItem
        key="7"
        label="7"
        value="7"
      />
      <PickerIOSItem
        key="8"
        label="8"
        value="8"
      />
      <PickerIOSItem
        key="9"
        label="9"
        value="9"
      />
      <PickerIOSItem
        key="10"
        label="10"
        value="10"
      />
      <PickerIOSItem
        key="11"
        label="11"
        value="11"
      />
      <PickerIOSItem
        key="12"
        label="12"
        value="12"
      />
      <PickerIOSItem
        key="13"
        label="13"
        value="13"
      />
      <PickerIOSItem
        key="14"
        label="14"
        value="14"
      />
      <PickerIOSItem
        key="15"
        label="15"
        value="15"
      />
      <PickerIOSItem
        key="16"
        label="16"
        value="16"
      />
      <PickerIOSItem
        key="17"
        label="17"
        value="17"
      />
      <PickerIOSItem
        key="18"
        label="18"
        value="18"
      />
      <PickerIOSItem
        key="19"
        label="19"
        value="19"
      />
      <PickerIOSItem
        key="20"
        label="20"
        value="20"
      />
      <PickerIOSItem
        key="21"
        label="21"
        value="21"
      />
      <PickerIOSItem
        key="22"
        label="22"
        value="22"
      />
      <PickerIOSItem
        key="23"
        label="23"
        value="23"
      />
      <PickerIOSItem
        key="24"
        label="24"
        value="24"
      />
      <PickerIOSItem
        key="25"
        label="25"
        value="25"
      />
      <PickerIOSItem
        key="26"
        label="26"
        value="26"
      />
      <PickerIOSItem
        key="27"
        label="27"
        value="27"
      />
      <PickerIOSItem
        key="28"
        label="28"
        value="28"
      />
      <PickerIOSItem
        key="29"
        label="29"
        value="29"
      />
      <PickerIOSItem
        key="30"
        label="30"
        value="30"
      />
      <PickerIOSItem
        key="31"
        label="31"
        value="31"
      />
      <PickerIOSItem
        key="32"
        label="32"
        value="32"
      />
      <PickerIOSItem
        key="33"
        label="33"
        value="33"
      />
      <PickerIOSItem
        key="34"
        label="34"
        value="34"
      />
      <PickerIOSItem
        key="35"
        label="35"
        value="35"
      />
      <PickerIOSItem
        key="36"
        label="36"
        value="36"
      />
      <PickerIOSItem
        key="37"
        label="37"
        value="37"
      />
      <PickerIOSItem
        key="38"
        label="38"
        value="38"
      />
      <PickerIOSItem
        key="39"
        label="39"
        value="39"
      />
      <PickerIOSItem
        key="40"
        label="40"
        value="40"
      />
      <PickerIOSItem
        key="41"
        label="41"
        value="41"
      />
      <PickerIOSItem
        key="42"
        label="42"
        value="42"
      />
      <PickerIOSItem
        key="43"
        label="43"
        value="43"
      />
      <PickerIOSItem
        key="44"
        label="44"
        value="44"
      />
      <PickerIOSItem
        key="45"
        label="45"
        value="45"
      />
      <PickerIOSItem
        key="46"
        label="46"
        value="46"
      />
      <PickerIOSItem
        key="47"
        label="47"
        value="47"
      />
      <PickerIOSItem
        key="48"
        label="48"
        value="48"
      />
      <PickerIOSItem
        key="49"
        label="49"
        value="49"
      />
      <PickerIOSItem
        key="50"
        label="50"
        value="50"
      />
      <PickerIOSItem
        key="51"
        label="51"
        value="51"
      />
      <PickerIOSItem
        key="52"
        label="52"
        value="52"
      />
      <PickerIOSItem
        key="53"
        label="53"
        value="53"
      />
      <PickerIOSItem
        key="54"
        label="54"
        value="54"
      />
      <PickerIOSItem
        key="55"
        label="55"
        value="55"
      />
      <PickerIOSItem
        key="56"
        label="56"
        value="56"
      />
      <PickerIOSItem
        key="57"
        label="57"
        value="57"
      />
      <PickerIOSItem
        key="58"
        label="58"
        value="58"
      />
      <PickerIOSItem
        key="59"
        label="59"
        value="59"
      />
      <PickerIOSItem
        key="60"
        label="60"
        value="60"
      />
      <PickerIOSItem
        key="61"
        label="61"
        value="61"
      />
      <PickerIOSItem
        key="62"
        label="62"
        value="62"
      />
      <PickerIOSItem
        key="63"
        label="63"
        value="63"
      />
      <PickerIOSItem
        key="64"
        label="64"
        value="64"
      />
      <PickerIOSItem
        key="65"
        label="65"
        value="65"
      />
      <PickerIOSItem
        key="66"
        label="66"
        value="66"
      />
      <PickerIOSItem
        key="67"
        label="67"
        value="67"
      />
      <PickerIOSItem
        key="68"
        label="68"
        value="68"
      />
      <PickerIOSItem
        key="69"
        label="69"
        value="69"
      />
      <PickerIOSItem
        key="70"
        label="70"
        value="70"
      />
      <PickerIOSItem
        key="71"
        label="71"
        value="71"
      />
      <PickerIOSItem
        key="72"
        label="72"
        value="72"
      />
      <PickerIOSItem
        key="73"
        label="73"
        value="73"
      />
      <PickerIOSItem
        key="74"
        label="74"
        value="74"
      />
      <PickerIOSItem
        key="75"
        label="75"
        value="75"
      />
      <PickerIOSItem
        key="76"
        label="76"
        value="76"
      />
      <PickerIOSItem
        key="77"
        label="77"
        value="77"
      />
      <PickerIOSItem
        key="78"
        label="78"
        value="78"
      />
      <PickerIOSItem
        key="79"
        label="79"
        value="79"
      />
      <PickerIOSItem
        key="80"
        label="80"
        value="80"
      />
      <PickerIOSItem
        key="81"
        label="81"
        value="81"
      />
      <PickerIOSItem
        key="82"
        label="82"
        value="82"
      />
      <PickerIOSItem
        key="83"
        label="83"
        value="83"
      />
      <PickerIOSItem
        key="84"
        label="84"
        value="84"
      />
      <PickerIOSItem
        key="85"
        label="85"
        value="85"
      />
      <PickerIOSItem
        key="86"
        label="86"
        value="86"
      />
      <PickerIOSItem
        key="87"
        label="87"
        value="87"
      />
      <PickerIOSItem
        key="88"
        label="88"
        value="88"
      />
      <PickerIOSItem
        key="89"
        label="89"
        value="89"
      />
      <PickerIOSItem
        key="90"
        label="90"
        value="90"
      />
      <PickerIOSItem
        key="91"
        label="91"
        value="91"
      />
      <PickerIOSItem
        key="92"
        label="92"
        value="92"
      />
      <PickerIOSItem
        key="93"
        label="93"
        value="93"
      />
      <PickerIOSItem
        key="94"
        label="94"
        value="94"
      />
      <PickerIOSItem
        key="95"
        label="95"
        value="95"
      />
      <PickerIOSItem
        key="96"
        label="96"
        value="96"
      />
      <PickerIOSItem
        key="97"
        label="97"
        value="97"
      />
      <PickerIOSItem
        key="98"
        label="98"
        value="98"
      />
      <PickerIOSItem
        key="99"
        label="99"
        value="99"
      />
      <PickerIOSItem
        key="100"
        label="100"
        value="100"
      />
    </ThemedPickerView>
    <TYText
      align={null}
      color={null}
      size={null}
      style={
        Object {
          "color": "#fff",
          "fontSize": 21,
          "left": 84,
          "position": "absolute",
        }
      }
      text="%"
      type={null}
      weight={null}
    />
  </View>
</View>
`;

exports[`TemperatureBrightPicker components componentWillReceiveProps render 1`] = `
<View
  disabled={false}
  hideThumb={false}
  maxBrightPercent={100}
  maxBrightness={1000}
  minBrightPercent={1}
  minBrightness={10}
  onChange={[Function]}
  onGrant={[Function]}
  onMove={[Function]}
  onPress={[Function]}
  onRelease={[Function]}
  stepBrightPercent={1}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <TemperaturePicker
    accessibilityLabel="TemperaturePicker"
    disabled={false}
    disalbedThumbOpacity={0.4}
    gradientBg={
      Array [
        Object {
          "offset": "0%",
          "stopColor": "#FFCA5C",
          "stopOpacity": 1,
        },
        Object {
          "offset": "50%",
          "stopColor": "#fff",
          "stopOpacity": 1,
        },
        Object {
          "offset": "100%",
          "stopColor": "#CDECFE",
          "stopOpacity": 1,
        },
      ]
    }
    hideThumb={false}
    innerRadius={132}
    max={1000}
    min={0}
    offsetAngle={180}
    onChange={[Function]}
    onGrant={[Function]}
    onMove={[Function]}
    onPress={[Function]}
    onRelease={[Function]}
    opacityAnimationDuration={150}
    opacityAnimationValue={1}
    radius={270}
    sideOffset={10}
    step={1}
    storageKey="temperature_bright_picker"
    thumbRadius={32}
    touchOffset={0}
    touchThumbRadius={0}
    value={1000}
  />
  <View
    style={
      Object {
        "height": 250.27984337537052,
        "justifyContent": "center",
        "position": "absolute",
        "width": 84,
      }
    }
  >
    <ThemedPickerView
      itemStyle={
        Object {
          "color": "#fff",
          "fontSize": 42,
          "height": 250.27984337537052,
        }
      }
      itemTextColor="#fff"
      onValueChange={[Function]}
      selectedItemTextColor="#fff"
      selectedValue="100"
      style={
        Object {
          "height": 250.27984337537052,
        }
      }
      textSize={42}
      theme={Object {}}
      useTYRCTPicker={false}
      visibleItemCount={3}
    >
      <PickerIOSItem
        key="1"
        label="1"
        value="1"
      />
      <PickerIOSItem
        key="2"
        label="2"
        value="2"
      />
      <PickerIOSItem
        key="3"
        label="3"
        value="3"
      />
      <PickerIOSItem
        key="4"
        label="4"
        value="4"
      />
      <PickerIOSItem
        key="5"
        label="5"
        value="5"
      />
      <PickerIOSItem
        key="6"
        label="6"
        value="6"
      />
      <PickerIOSItem
        key="7"
        label="7"
        value="7"
      />
      <PickerIOSItem
        key="8"
        label="8"
        value="8"
      />
      <PickerIOSItem
        key="9"
        label="9"
        value="9"
      />
      <PickerIOSItem
        key="10"
        label="10"
        value="10"
      />
      <PickerIOSItem
        key="11"
        label="11"
        value="11"
      />
      <PickerIOSItem
        key="12"
        label="12"
        value="12"
      />
      <PickerIOSItem
        key="13"
        label="13"
        value="13"
      />
      <PickerIOSItem
        key="14"
        label="14"
        value="14"
      />
      <PickerIOSItem
        key="15"
        label="15"
        value="15"
      />
      <PickerIOSItem
        key="16"
        label="16"
        value="16"
      />
      <PickerIOSItem
        key="17"
        label="17"
        value="17"
      />
      <PickerIOSItem
        key="18"
        label="18"
        value="18"
      />
      <PickerIOSItem
        key="19"
        label="19"
        value="19"
      />
      <PickerIOSItem
        key="20"
        label="20"
        value="20"
      />
      <PickerIOSItem
        key="21"
        label="21"
        value="21"
      />
      <PickerIOSItem
        key="22"
        label="22"
        value="22"
      />
      <PickerIOSItem
        key="23"
        label="23"
        value="23"
      />
      <PickerIOSItem
        key="24"
        label="24"
        value="24"
      />
      <PickerIOSItem
        key="25"
        label="25"
        value="25"
      />
      <PickerIOSItem
        key="26"
        label="26"
        value="26"
      />
      <PickerIOSItem
        key="27"
        label="27"
        value="27"
      />
      <PickerIOSItem
        key="28"
        label="28"
        value="28"
      />
      <PickerIOSItem
        key="29"
        label="29"
        value="29"
      />
      <PickerIOSItem
        key="30"
        label="30"
        value="30"
      />
      <PickerIOSItem
        key="31"
        label="31"
        value="31"
      />
      <PickerIOSItem
        key="32"
        label="32"
        value="32"
      />
      <PickerIOSItem
        key="33"
        label="33"
        value="33"
      />
      <PickerIOSItem
        key="34"
        label="34"
        value="34"
      />
      <PickerIOSItem
        key="35"
        label="35"
        value="35"
      />
      <PickerIOSItem
        key="36"
        label="36"
        value="36"
      />
      <PickerIOSItem
        key="37"
        label="37"
        value="37"
      />
      <PickerIOSItem
        key="38"
        label="38"
        value="38"
      />
      <PickerIOSItem
        key="39"
        label="39"
        value="39"
      />
      <PickerIOSItem
        key="40"
        label="40"
        value="40"
      />
      <PickerIOSItem
        key="41"
        label="41"
        value="41"
      />
      <PickerIOSItem
        key="42"
        label="42"
        value="42"
      />
      <PickerIOSItem
        key="43"
        label="43"
        value="43"
      />
      <PickerIOSItem
        key="44"
        label="44"
        value="44"
      />
      <PickerIOSItem
        key="45"
        label="45"
        value="45"
      />
      <PickerIOSItem
        key="46"
        label="46"
        value="46"
      />
      <PickerIOSItem
        key="47"
        label="47"
        value="47"
      />
      <PickerIOSItem
        key="48"
        label="48"
        value="48"
      />
      <PickerIOSItem
        key="49"
        label="49"
        value="49"
      />
      <PickerIOSItem
        key="50"
        label="50"
        value="50"
      />
      <PickerIOSItem
        key="51"
        label="51"
        value="51"
      />
      <PickerIOSItem
        key="52"
        label="52"
        value="52"
      />
      <PickerIOSItem
        key="53"
        label="53"
        value="53"
      />
      <PickerIOSItem
        key="54"
        label="54"
        value="54"
      />
      <PickerIOSItem
        key="55"
        label="55"
        value="55"
      />
      <PickerIOSItem
        key="56"
        label="56"
        value="56"
      />
      <PickerIOSItem
        key="57"
        label="57"
        value="57"
      />
      <PickerIOSItem
        key="58"
        label="58"
        value="58"
      />
      <PickerIOSItem
        key="59"
        label="59"
        value="59"
      />
      <PickerIOSItem
        key="60"
        label="60"
        value="60"
      />
      <PickerIOSItem
        key="61"
        label="61"
        value="61"
      />
      <PickerIOSItem
        key="62"
        label="62"
        value="62"
      />
      <PickerIOSItem
        key="63"
        label="63"
        value="63"
      />
      <PickerIOSItem
        key="64"
        label="64"
        value="64"
      />
      <PickerIOSItem
        key="65"
        label="65"
        value="65"
      />
      <PickerIOSItem
        key="66"
        label="66"
        value="66"
      />
      <PickerIOSItem
        key="67"
        label="67"
        value="67"
      />
      <PickerIOSItem
        key="68"
        label="68"
        value="68"
      />
      <PickerIOSItem
        key="69"
        label="69"
        value="69"
      />
      <PickerIOSItem
        key="70"
        label="70"
        value="70"
      />
      <PickerIOSItem
        key="71"
        label="71"
        value="71"
      />
      <PickerIOSItem
        key="72"
        label="72"
        value="72"
      />
      <PickerIOSItem
        key="73"
        label="73"
        value="73"
      />
      <PickerIOSItem
        key="74"
        label="74"
        value="74"
      />
      <PickerIOSItem
        key="75"
        label="75"
        value="75"
      />
      <PickerIOSItem
        key="76"
        label="76"
        value="76"
      />
      <PickerIOSItem
        key="77"
        label="77"
        value="77"
      />
      <PickerIOSItem
        key="78"
        label="78"
        value="78"
      />
      <PickerIOSItem
        key="79"
        label="79"
        value="79"
      />
      <PickerIOSItem
        key="80"
        label="80"
        value="80"
      />
      <PickerIOSItem
        key="81"
        label="81"
        value="81"
      />
      <PickerIOSItem
        key="82"
        label="82"
        value="82"
      />
      <PickerIOSItem
        key="83"
        label="83"
        value="83"
      />
      <PickerIOSItem
        key="84"
        label="84"
        value="84"
      />
      <PickerIOSItem
        key="85"
        label="85"
        value="85"
      />
      <PickerIOSItem
        key="86"
        label="86"
        value="86"
      />
      <PickerIOSItem
        key="87"
        label="87"
        value="87"
      />
      <PickerIOSItem
        key="88"
        label="88"
        value="88"
      />
      <PickerIOSItem
        key="89"
        label="89"
        value="89"
      />
      <PickerIOSItem
        key="90"
        label="90"
        value="90"
      />
      <PickerIOSItem
        key="91"
        label="91"
        value="91"
      />
      <PickerIOSItem
        key="92"
        label="92"
        value="92"
      />
      <PickerIOSItem
        key="93"
        label="93"
        value="93"
      />
      <PickerIOSItem
        key="94"
        label="94"
        value="94"
      />
      <PickerIOSItem
        key="95"
        label="95"
        value="95"
      />
      <PickerIOSItem
        key="96"
        label="96"
        value="96"
      />
      <PickerIOSItem
        key="97"
        label="97"
        value="97"
      />
      <PickerIOSItem
        key="98"
        label="98"
        value="98"
      />
      <PickerIOSItem
        key="99"
        label="99"
        value="99"
      />
      <PickerIOSItem
        key="100"
        label="100"
        value="100"
      />
    </ThemedPickerView>
    <TYText
      align={null}
      color={null}
      size={null}
      style={
        Object {
          "color": "#fff",
          "fontSize": 21,
          "left": 84,
          "position": "absolute",
        }
      }
      text="%"
      type={null}
      weight={null}
    />
  </View>
</View>
`;

exports[`TemperatureBrightPicker components method 1`] = `
<View
  maxBrightPercent={100}
  maxBrightness={1000}
  minBrightPercent={1}
  minBrightness={10}
  onChange={[Function]}
  onGrant={[Function]}
  onMove={[Function]}
  onPress={[Function]}
  onRelease={[Function]}
  stepBrightPercent={1}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <TemperaturePicker
    accessibilityLabel="TemperaturePicker"
    disabled={false}
    disalbedThumbOpacity={0.4}
    gradientBg={
      Array [
        Object {
          "offset": "0%",
          "stopColor": "#FFCA5C",
          "stopOpacity": 1,
        },
        Object {
          "offset": "50%",
          "stopColor": "#fff",
          "stopOpacity": 1,
        },
        Object {
          "offset": "100%",
          "stopColor": "#CDECFE",
          "stopOpacity": 1,
        },
      ]
    }
    hideThumb={false}
    innerRadius={132}
    max={1000}
    min={0}
    offsetAngle={180}
    onChange={[Function]}
    onGrant={[Function]}
    onMove={[Function]}
    onPress={[Function]}
    onRelease={[Function]}
    opacityAnimationDuration={150}
    opacityAnimationValue={1}
    radius={270}
    sideOffset={10}
    step={1}
    storageKey="temperature_bright_picker"
    thumbRadius={32}
    touchOffset={0}
    touchThumbRadius={0}
    value={1000}
  />
  <View
    style={
      Object {
        "height": 250.27984337537052,
        "justifyContent": "center",
        "position": "absolute",
        "width": 84,
      }
    }
  >
    <ThemedPickerView
      itemStyle={
        Object {
          "color": "#fff",
          "fontSize": 42,
          "height": 250.27984337537052,
        }
      }
      itemTextColor="#fff"
      onValueChange={[Function]}
      selectedItemTextColor="#fff"
      selectedValue="100"
      style={
        Object {
          "height": 250.27984337537052,
        }
      }
      textSize={42}
      theme={Object {}}
      useTYRCTPicker={false}
      visibleItemCount={3}
    >
      <PickerIOSItem
        key="1"
        label="1"
        value="1"
      />
      <PickerIOSItem
        key="2"
        label="2"
        value="2"
      />
      <PickerIOSItem
        key="3"
        label="3"
        value="3"
      />
      <PickerIOSItem
        key="4"
        label="4"
        value="4"
      />
      <PickerIOSItem
        key="5"
        label="5"
        value="5"
      />
      <PickerIOSItem
        key="6"
        label="6"
        value="6"
      />
      <PickerIOSItem
        key="7"
        label="7"
        value="7"
      />
      <PickerIOSItem
        key="8"
        label="8"
        value="8"
      />
      <PickerIOSItem
        key="9"
        label="9"
        value="9"
      />
      <PickerIOSItem
        key="10"
        label="10"
        value="10"
      />
      <PickerIOSItem
        key="11"
        label="11"
        value="11"
      />
      <PickerIOSItem
        key="12"
        label="12"
        value="12"
      />
      <PickerIOSItem
        key="13"
        label="13"
        value="13"
      />
      <PickerIOSItem
        key="14"
        label="14"
        value="14"
      />
      <PickerIOSItem
        key="15"
        label="15"
        value="15"
      />
      <PickerIOSItem
        key="16"
        label="16"
        value="16"
      />
      <PickerIOSItem
        key="17"
        label="17"
        value="17"
      />
      <PickerIOSItem
        key="18"
        label="18"
        value="18"
      />
      <PickerIOSItem
        key="19"
        label="19"
        value="19"
      />
      <PickerIOSItem
        key="20"
        label="20"
        value="20"
      />
      <PickerIOSItem
        key="21"
        label="21"
        value="21"
      />
      <PickerIOSItem
        key="22"
        label="22"
        value="22"
      />
      <PickerIOSItem
        key="23"
        label="23"
        value="23"
      />
      <PickerIOSItem
        key="24"
        label="24"
        value="24"
      />
      <PickerIOSItem
        key="25"
        label="25"
        value="25"
      />
      <PickerIOSItem
        key="26"
        label="26"
        value="26"
      />
      <PickerIOSItem
        key="27"
        label="27"
        value="27"
      />
      <PickerIOSItem
        key="28"
        label="28"
        value="28"
      />
      <PickerIOSItem
        key="29"
        label="29"
        value="29"
      />
      <PickerIOSItem
        key="30"
        label="30"
        value="30"
      />
      <PickerIOSItem
        key="31"
        label="31"
        value="31"
      />
      <PickerIOSItem
        key="32"
        label="32"
        value="32"
      />
      <PickerIOSItem
        key="33"
        label="33"
        value="33"
      />
      <PickerIOSItem
        key="34"
        label="34"
        value="34"
      />
      <PickerIOSItem
        key="35"
        label="35"
        value="35"
      />
      <PickerIOSItem
        key="36"
        label="36"
        value="36"
      />
      <PickerIOSItem
        key="37"
        label="37"
        value="37"
      />
      <PickerIOSItem
        key="38"
        label="38"
        value="38"
      />
      <PickerIOSItem
        key="39"
        label="39"
        value="39"
      />
      <PickerIOSItem
        key="40"
        label="40"
        value="40"
      />
      <PickerIOSItem
        key="41"
        label="41"
        value="41"
      />
      <PickerIOSItem
        key="42"
        label="42"
        value="42"
      />
      <PickerIOSItem
        key="43"
        label="43"
        value="43"
      />
      <PickerIOSItem
        key="44"
        label="44"
        value="44"
      />
      <PickerIOSItem
        key="45"
        label="45"
        value="45"
      />
      <PickerIOSItem
        key="46"
        label="46"
        value="46"
      />
      <PickerIOSItem
        key="47"
        label="47"
        value="47"
      />
      <PickerIOSItem
        key="48"
        label="48"
        value="48"
      />
      <PickerIOSItem
        key="49"
        label="49"
        value="49"
      />
      <PickerIOSItem
        key="50"
        label="50"
        value="50"
      />
      <PickerIOSItem
        key="51"
        label="51"
        value="51"
      />
      <PickerIOSItem
        key="52"
        label="52"
        value="52"
      />
      <PickerIOSItem
        key="53"
        label="53"
        value="53"
      />
      <PickerIOSItem
        key="54"
        label="54"
        value="54"
      />
      <PickerIOSItem
        key="55"
        label="55"
        value="55"
      />
      <PickerIOSItem
        key="56"
        label="56"
        value="56"
      />
      <PickerIOSItem
        key="57"
        label="57"
        value="57"
      />
      <PickerIOSItem
        key="58"
        label="58"
        value="58"
      />
      <PickerIOSItem
        key="59"
        label="59"
        value="59"
      />
      <PickerIOSItem
        key="60"
        label="60"
        value="60"
      />
      <PickerIOSItem
        key="61"
        label="61"
        value="61"
      />
      <PickerIOSItem
        key="62"
        label="62"
        value="62"
      />
      <PickerIOSItem
        key="63"
        label="63"
        value="63"
      />
      <PickerIOSItem
        key="64"
        label="64"
        value="64"
      />
      <PickerIOSItem
        key="65"
        label="65"
        value="65"
      />
      <PickerIOSItem
        key="66"
        label="66"
        value="66"
      />
      <PickerIOSItem
        key="67"
        label="67"
        value="67"
      />
      <PickerIOSItem
        key="68"
        label="68"
        value="68"
      />
      <PickerIOSItem
        key="69"
        label="69"
        value="69"
      />
      <PickerIOSItem
        key="70"
        label="70"
        value="70"
      />
      <PickerIOSItem
        key="71"
        label="71"
        value="71"
      />
      <PickerIOSItem
        key="72"
        label="72"
        value="72"
      />
      <PickerIOSItem
        key="73"
        label="73"
        value="73"
      />
      <PickerIOSItem
        key="74"
        label="74"
        value="74"
      />
      <PickerIOSItem
        key="75"
        label="75"
        value="75"
      />
      <PickerIOSItem
        key="76"
        label="76"
        value="76"
      />
      <PickerIOSItem
        key="77"
        label="77"
        value="77"
      />
      <PickerIOSItem
        key="78"
        label="78"
        value="78"
      />
      <PickerIOSItem
        key="79"
        label="79"
        value="79"
      />
      <PickerIOSItem
        key="80"
        label="80"
        value="80"
      />
      <PickerIOSItem
        key="81"
        label="81"
        value="81"
      />
      <PickerIOSItem
        key="82"
        label="82"
        value="82"
      />
      <PickerIOSItem
        key="83"
        label="83"
        value="83"
      />
      <PickerIOSItem
        key="84"
        label="84"
        value="84"
      />
      <PickerIOSItem
        key="85"
        label="85"
        value="85"
      />
      <PickerIOSItem
        key="86"
        label="86"
        value="86"
      />
      <PickerIOSItem
        key="87"
        label="87"
        value="87"
      />
      <PickerIOSItem
        key="88"
        label="88"
        value="88"
      />
      <PickerIOSItem
        key="89"
        label="89"
        value="89"
      />
      <PickerIOSItem
        key="90"
        label="90"
        value="90"
      />
      <PickerIOSItem
        key="91"
        label="91"
        value="91"
      />
      <PickerIOSItem
        key="92"
        label="92"
        value="92"
      />
      <PickerIOSItem
        key="93"
        label="93"
        value="93"
      />
      <PickerIOSItem
        key="94"
        label="94"
        value="94"
      />
      <PickerIOSItem
        key="95"
        label="95"
        value="95"
      />
      <PickerIOSItem
        key="96"
        label="96"
        value="96"
      />
      <PickerIOSItem
        key="97"
        label="97"
        value="97"
      />
      <PickerIOSItem
        key="98"
        label="98"
        value="98"
      />
      <PickerIOSItem
        key="99"
        label="99"
        value="99"
      />
      <PickerIOSItem
        key="100"
        label="100"
        value="100"
      />
    </ThemedPickerView>
    <TYText
      align={null}
      color={null}
      size={null}
      style={
        Object {
          "color": "#fff",
          "fontSize": 21,
          "left": 84,
          "position": "absolute",
        }
      }
      text="%"
      type={null}
      weight={null}
    />
  </View>
</View>
`;
